plugins/auto-sizes/hooks.php

```diff
@@ -203,7 +203,16 @@ function auto_sizes_improve_image_sizes_attributes( string $content ): string {
 	}
 
 	if ( is_string( $sizes ) ) {
+		// Add the filter to update the sizes calculations to "wp_calculate_image_sizes".
+		$filter = static function () use ( $sizes ): string {
+			return $sizes;
+		};
+		add_filter( 'wp_calculate_image_sizes', $filter );
+
 		$processor->set_attribute( 'sizes', $sizes );
+
+		// Remove the filter to update the sizes calculations to "wp_calculate_image_sizes".
+		remove_filter( 'wp_calculate_image_sizes', $filter );
```

Comment:
I don't see how this code will have any effect. The `wp_calculate_image_sizes` filter is applied when `wp_calculate_image_sizes()` is called. But `wp_calculate_image_sizes()` is not called here. The only code being called when the filter is applied is `$processor->set_attribute( 'sizes', $sizes )`, and this does not have any side effects. I don't see how this filter is supposed to fix things.

Reply:
In other words, with these changes removed, the test you added still passes. So the test doesn't seem to be checking the problem.

Reply:
I have pushed the changes that show the failed tests: https://github.com/WordPress/performance/actions/runs/10089272264/job/27896437786?pr=1382#step:9:32.

What I want to achieve with this PR:
- When someone uses the core `wp_calculate_image_sizes()`, it should return core sizes, not the improved sizes.
- When the image is filtered through the improved sizes filter `add_filter( 'wp_content_img_tag', 'auto_sizes_improve_image_sizes_attributes', 9 );`, the core `wp_calculate_image_sizes()` should return the new sizes, not the core sizes.

This PR is a proof of concept, but I'm happy to update the approach.

Reply:
What I mean is that there should be tests that fail when this added `$filter` code is removed. As it stands, I don't see how these lines will have any effect. So I guess that is reflected in the new tests that now fail?